packages/plugin-ext/src/plugin/documents.ts

```diff
@@ -247,10 +247,10 @@ export class DocumentsExtImpl implements DocumentsExt {
             if (options.selection) {
                 const { start, end } = options.selection;
                 selection = {
-                    startLineNumber: start.line,
```

Comment:
I don't think it is a good place, it should be rather fixed here for all clients: https://github.com/siemens/theia/blob/8f5b57a34e9d063908c1948abc8f72a602d50b9f/packages/plugin-ext/src/main/browser/documents-main.ts#L241-L244

I'm not sure why we subtract there. I think we should do nothing. VS Code API is zero based and Theia editor manager api are also zero based.

We for sure should have a VS Code extension to test for 3 paths:
- showTestDocument
- vscode.open with selection
- vscode.openDiff with selection

We should verify that vscode.open and vscode.openDiff accept zero based positions.

Reply:
True, I agree the way to go here is to always count from 0 to be consistent with what VS Code does. The values here do count from 1 so we should probably change all these instances to count from 0 instead, e.g. the place you linked and also update [the documentation](https://github.com/eclipse-theia/theia/blob/5ea335c560573de3d176c64e37fa18b477434d81/packages/plugin-ext/src/common/plugin-api-rpc-model.ts#L52).

Will take a look at this and provide a simple demonstration later.